src/Dropdown/Dropdown.js

```diff
@@ -121,45 +122,42 @@ export default class Dropdown extends React.Component {
     }
   };
 
-  attachAutocloseListener() {
-    if (!this.props.autoclose) {
-      return;
-    }
-    document.addEventListener('click', this.autoclose);
-  }
-
-  detachAutocloseListener() {
-    document.removeEventListener('click', this.autoclose);
-  }
-
   autoclose = e => {
-    if (this.state.isOpen && !this.menuRef.current.contains(e.target)) {
+    if (this.state.isOpen) {
       this.hide();
     }
   };
 
+  handleBlur = () => {
+    this.timeoutId = setTimeout(this.autoclose, 500);
```

Comment:
500 feels a little long in testing - let's go for 175 or so.

Reply:
Delay removed.

Reply:
175 delay restored for mouseenter/mouseleave support.